### PR TITLE
Disallow empty statements made by semicolons

### PR DESCRIPTION
**Pull request by Burgos**
This turns the previous warning when using a semicolon
for an empty statement to a compilation error.

### CODE REVIEW

File: src/ddmd/parse.d [hunk 1]
```diff
@@ -4664,7 +4664,7 @@ final class Parser(AST) : Lexer
         case TOKlcurly:
             if (f.frequire || f.fensure)
                 error("missing body { ... } after in or out");
-            f.fbody = parseStatement(PSsemi);
+            f.fbody = parseStatement(0);
```
Comment (Burgos) [inline src/ddmd/parse.d]: I was going to make a new enum value `PSdefault` or so, but I saw `parseStatement` called with `0` in other places as well, so I left it like this.

File: src/ddmd/parse.d [hunk 2]
```diff
@@ -5112,10 +5112,7 @@ final class Parser(AST) : Lexer
         case TOKsemicolon:
             if (!(flags & PSsemi_ok))
             {
-                if (flags & PSsemi)
-                    warning(loc, "use '{ }' for an empty statement, not a ';'");
-                else
-                    error("use '{ }' for an empty statement, not a ';'");
+                error("use '{ }' for an empty statement, not a ';'");
```
Comment (BBasile) [inline src/ddmd/parse.d]: You cant pass from warning to error. There must be a  deprecation period before.

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by WalterBright** [comment]
This must go through a period of time where it is deprecated first.

**Comment by Burgos** [comment]
Ah, true. I got mislead by the compiler's source, which states this is deprecated (See the `PSsemi`'s ddoc). Will fix, thanks!

**Comment by Burgos** [comment]
Changed to a deprecation.

**Review by WalterBright — APPROVED** [review]
Nice work!

**Comment by Burgos** [comment]
I'm not sure what's up with Jenkins, but he also failed in the trivial #6891 PR

**Comment by wilzbach** [comment]
> I'm not sure what's up with Jenkins, but he also failed in the trivial #6891 PR

Yes, the Jenkins server was down for a week and now we are playing goose hunting for the introduced regressions in downstream projects, see e.g. https://github.com/dlang/phobos/pull/5469
tl;dr: don't worry about it, the errors should be fixed soon.

**Comment by nemanja-boric-sociomantic** [comment]
All clear, thanks!
